src/ApiCallable.php

```diff
@@ -121,9 +121,12 @@ private static function setCustomHeader($callable, $headerDescriptor)
                 throw new InvalidArgumentException('Metadata argument is not found.');
             } else {
                 $metadata = $params[self::GRPC_CALLABLE_METADATA_INDEX];
-                $params[self::GRPC_CALLABLE_METADATA_INDEX] =
-                    array_merge($metadata, $headerDescriptor->getHeader());
-                call_user_func_array($callable, $params);
+                $headers = $headerDescriptor->getHeader();
+                if (array_key_exists('headers', $metadata)) {
+                    $headers = array_merge($headers, $metadata['headers']);
```

Comment:
Our generated wrappers won't provide a way to wrapper callers to override it; it might be handy for our wrappers themselves to be able to override it though. It might be worth clarifying this in the docs of ApiCallable, in any case.